### PR TITLE
Filter component/484 

### PR DESCRIPTION
**Pull request by jul4ajka**
#484 

Created custom component for filtering, search, sorting and clear filters.

### CODE REVIEW

File: src/pages/categories/categories.js [hunk 1]
```diff
@@ -37,24 +45,34 @@ const Categories = () => {
...
-      .sort((a, b) => {
-        if (a.name[0].value.toLowerCase() > b.name[0].value.toLowerCase()) {
-          return 1;
-        }
-        if (a.name[0].value.toLowerCase() < b.name[0].value.toLowerCase()) {
-          return -1;
-        }
-        return 0;
-      })
+        // .sort((a, b) => {
```
Comment (AndriyLymych) [inline src/pages/categories/categories.js:67]: remove commented code

File: src/components/filter-search-sort/nav-filter-item/nav-filter-item.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  filterValues,
+  filterList,
+  filterHandler
+}) => {
+  const styles = useStyles();
+
+  const formGroupOptions = filterList.map((item, idx) => (
+    <MenuItem data-cy={`user-filters-list-${item}`} key={item} value={item}>
+      <Checkbox
+        checked={filterValues.findIndex((filter) => filter === item) !== -1}
```
Comment (moran711) [inline src/components/filter-search-sort/nav-filter-item/nav-filter-item.js:33]: You can use https://lodash.com/docs/4.17.15#findIndex

File: src/components/filter-search-sort/nav-filter-item/nav-filter-item.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+const NavFilterItem = ({
+  buttonName,
+  filterLabels,
+  filterValues,
+  filterList,
+  filterHandler
+}) => {
+  const styles = useStyles();
+
+  const formGroupOptions = filterList.map((item, idx) => (
```
Comment (moran711) [inline src/components/filter-search-sort/nav-filter-item/nav-filter-item.js:30]: Please use `useMemo `hook and put `filterList `in dependency

Comment (moran711) [reply src/components/filter-search-sort/nav-filter-item/nav-filter-item.js:30]: You can use `map `method from lodash

File: src/components/filter-search-sort/nav-filters/nav-filters.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import NavFilterItem from '../nav-filter-item';
+
+const NavFilters = ({ filterOptions }) =>
+  Object.values(
+    filterOptions
```
Comment (moran711) [inline src/components/filter-search-sort/nav-filters/nav-filters.js:7]: same

File: src/components/filter-search-sort/nav-search/nav-search.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    setSearchValue(event.target.value);
+  };
+
+  const handleSearchSubmit = (event) => {
+    if (event.key === submitKey) {
+      handleUserSearch();
+    }
+  };
+
+  const handleUserSearch = () => {
```
Comment (moran711) [inline src/components/filter-search-sort/nav-search/nav-search.js:33]: Use `useCallback `hook for this)

File: src/components/filter-search-sort/nav-sort/nav-sort.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import { FormControl, MenuItem, Select } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import { useStyles } from './nav-sort.styles';
+
+const NavSort = ({ sortOptions }) => {
+  const [sortValue, setSortValue] = useState(sortOptions.labels[1].value);
+  const styles = useStyles();
+  const { setSorting } = sortOptions;
+
+  const selectOptions = sortOptions.labels.map(({ label, value }) => (
```
Comment (moran711) [inline src/components/filter-search-sort/nav-sort/nav-sort.js:11]: same

File: src/components/filter-search-sort/nav-sort/nav-sort.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    <MenuItem key={label} value={value}>
+      {label}
+    </MenuItem>
+  ));
+
+  const selectHandler = (e) => {
```
Comment (moran711) [inline src/components/filter-search-sort/nav-sort/nav-sort.js:17]: use `useCallback `hook for this)

**Pull request merged**